### PR TITLE
Replace `flake8`, `isort`, `pyupgrade` with `ruff`

### PR DESCRIPTION
**Pull request by kaxil**
https://github.com/charliermarsh/ruff/ is a faster replacement of most of the linting tools we use, and is starting to picked up by several other projects. Even project like Pandas have adopted Ruff (https://github.com/pandas-dev/pandas/pull/50160)

This PR replaces flake8, isort, pyupgrade.

```
❯ time pre-commit run flake8 --all-files
flake8...................................................................Passed
pre-commit run flake8 --all-files  3.48s user 0.55s system 372% cpu 1.084 total

❯ time pre-commit run isort --all-files
Run isort................................................................Passed
pre-commit run isort --all-files  0.31s user 0.18s system 50% cpu 0.973 total

❯ time pre-commit run pyupgrade --all-files
pyupgrade................................................................Passed
pre-commit run pyupgrade --all-files  1.26s user 0.23s system 280% cpu 0.530 total
```
 vs
```
❯ time pre-commit run ruff --all-files
ruff.....................................................................Passed
pre-commit run ruff --all-files  0.37s user 0.16s system 142% cpu 0.373 total
```

**_Drops from 5s to 0.3s_**

Some other popular tools that have adopted it:

- [FastAPI](https://github.com/tiangolo/fastapi)
- [Bokeh](https://github.com/bokeh/bokeh)
- [Zulip](https://github.com/zulip/zulip)
- [Pydantic](https://github.com/pydantic/pydantic)
- [Sphinx](https://github.com/sphinx-doc/sphinx)
- [Hatch](https://github.com/pypa/hatch)
- [Jupyter](https://github.com/jupyter-server/jupyter_server)
- [Synapse](https://github.com/matrix-org/synapse)
- [Saleor](https://github.com/saleor/saleor)
- [Polars](https://github.com/pola-rs/polars)
- [Ibis](https://github.com/ibis-project/ibis)
- [OpenBB](https://github.com/OpenBB-finance/OpenBBTerminal)

It will also be used by Apache Airflow :) and should be used in Astronomer-providers too.

### CODE REVIEW

**Review by feluelle — APPROVED** [review]
🚀

**Review by tatiana — APPROVED** [review]
It's very impressive the time improvement, @kaxil !
Do you see any drawbacks to the adoption of ruff?

**Comment by kaxil** [comment]
> It's very impressive the time improvement, @kaxil ! Do you see any drawbacks to the adoption of ruff?

It is bleeding edge -- maybe debugging non-trivial error might coz an issue as it is written in Rust 🤷 -- We will find out :D

File: python-sdk/src/astro/sql/operators/raw_sql.py [hunk 1]
```diff
@@ -7,8 +7,8 @@
 try:
     from airflow.decorators.base import TaskDecorator, task_decorator_factory
 except ImportError:
-    from airflow.decorators.base import task_decorator_factory
     from airflow.decorators import _TaskDecorator as TaskDecorator
+    from airflow.decorators.base import task_decorator_factory
```
[the remark this answers is not in the document]
Comment (kaxil) [reply python-sdk/src/astro/sql/operators/raw_sql.py:11]: Good spot, updated

**Review by pankajkoti — APPROVED** [review]

**Pull request merged**